full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts

```diff
@@ -0,0 +1,306 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { IModelConnection, SnapshotConnection } from "@bentley/imodeljs-frontend";
+import { ChildNodeSpecificationTypes, Ruleset, RuleTypes } from "@bentley/presentation-common";
+import { Presentation } from "@bentley/presentation-frontend";
+import { initialize, terminate } from "../IntegrationTests";
+
+describe("Learning Snippets", () => {
+
+  describe("Ruleset Variables", () => {
+
+    let imodel: IModelConnection;
+
+    beforeEach(async () => {
+      await initialize();
+      imodel = await SnapshotConnection.openFile("assets/datasets/Properties_60InstancesWithUrl2.ibim");
+    });
+
+    afterEach(async () => {
+      await imodel.close();
+      await terminate();
+    });
+
+    it("uses ruleset variable in rule condition", async () => {
+      // __PUBLISH_EXTRACT_START__ RulesetVariables.InRuleCondition.Ruleset
+      // The ruleset has two root node rules - one for models and one for elements. The one actually used
+      // depends on a ruleset variable value, which can be changed without modifying the ruleset itself.
+      const variableId = "root-nodes-type";
+      const ruleset: Ruleset = {
+        id: "test",
+        rules: [{
+          ruleType: RuleTypes.RootNodes,
+          condition: `GetVariableStringValue("${variableId}") = "models"`,
+          specifications: [{
+            specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+            classes: { schemaName: "BisCore", classNames: ["Model"] },
+            arePolymorphic: true,
+          }],
+        }, {
+          ruleType: RuleTypes.RootNodes,
+          condition: `GetVariableStringValue("${variableId}") = "elements"`,
+          specifications: [{
+            specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+            classes: { schemaName: "BisCore", classNames: ["Element"] },
+            arePolymorphic: true,
+          }],
+        }],
+      };

```

Comment:
As someone encountering ruleset variables for the first time, I had to stop and think whether "${variableId}" on line 36 is a part of rules variable syntax. Of course, they aren't, but I think the examples would be more obvious if string templating was avoided. Also, I would give this ruleset variable another id: `"var-tree-type"`.

Reply:
Agree - hardcoded values are more obvious to understand. Note: I didn't prefix variable ID with `var-` - seems unnecessary to me.

Reply:
Thanks. Maybe then call out the name of the variable in the preceding text so the reader knows what to pay attention to?

Reply:
- Made the IDs `UPPER_CASE`, which, I think, exposes them more.
- Added comments next to the ruleset definitions explaining what is controlled by which ruleset variable.

Reply:
Looks great now, thanks.